src/cryptography/sign.js

```diff
@@ -172,12 +172,19 @@ export const signAndPrintMessage = (message, passphrase, secondPassphrase) => {
 	return printSignedMessage(signedMessage);
 };
 
-export const signData = (data, passphrase) => {
+export const signDataWithPassphrase = (data, passphrase) => {
 	const { privateKey } = getPrivateAndPublicKeyBytesFromPassphrase(passphrase);
 	const signature = naclInstance.crypto_sign_detached(data, privateKey);
 	return bufferToHex(signature);
 };
 
+export const signDataWithPrivateKey = (data, privateKey) => {
+	const signature = naclInstance.crypto_sign_detached(data, privateKey);
+	return bufferToHex(signature);
+};
+
+export const signData = signDataWithPassphrase;
```

Comment:
I think this `signData` adds more confusion than it is helping. You never know if it was the password or the key version and you have multiple methods doing the same thing. Especially with no strict types in place, it is too easy to mix up methods and input types.

Isn't 1.0.0 a great chance to break the existing function and add to the migration guide:

* signData was renamed to signDataWithPassphrase

Reply:
It is up to core developer, but I agree with your decision. What do you think @willclarktech, @Tosch110?